Permission denied error when installing plugin from update center

### Cookbook version
5.0.3

### Chef-client version
13.2.20

### Platform Details
platform=ubuntu
platform_version=14.04


### Scenario:
When attempting to install plugins through `jenkins_plugin` I get an `java.io.FileNotFoundException: /var/chef/cache/greenballs-1.15.plugin (Permission denied)` error.
Plugin file permissions are set like this by `remote_file`:
`-rw-r----- 1 root root  111507 Jul 20 09:56 greenballs-1.15.plugin`

Granting read access to everyone or making jenkins owner of the file solves the problem.

### Steps to Reproduce:
```
apt_package 'jenkins'
include_recipe "jenkins::master"
jenkins_plugin 'greenballs' do
  action :install
end
```

Don't think if it's related, but we're using `oracle-java8-installer` package from `ppa:webupd8team/java` for installing Java 8 since there are no official packages for Ubuntu 14.04

### Expected Result:
Plugin installed.

### Actual Result:
```
---- Begin output of "java" -jar "/var/chef/cache/jenkins-cli.jar" -s http://localhost:8080 -"remoting" install-plugin file:///var/chef/cache/greenballs-1.15.plugin -name greenballs  ----
STDOUT: Installing a plugin from file:/var/chef/cache/greenballs-1.15.plugin
STDERR: ERROR: Unexpected exception occurred while performing install-plugin command.
  java.io.FileNotFoundException: /var/chef/cache/greenballs-1.15.plugin (Permission denied)
  at java.io.FileInputStream.open0(Native Method)
  at java.io.FileInputStream.open(FileInputStream.java:195)
  at java.io.FileInputStream.<init>(FileInputStream.java:138)
  at java.io.FileInputStream.<init>(FileInputStream.java:93)
  at sun.net.www.protocol.file.FileURLConnection.connect(FileURLConnection.java:90)
  at sun.net.www.protocol.file.FileURLConnection.getInputStream(FileURLConnection.java:188)
  at java.net.URL.openStream(URL.java:1045)
  at hudson.FilePath.copyFrom(FilePath.java:890)
  at hudson.cli.InstallPluginCommand.run(InstallPluginCommand.java:127)
  at hudson.cli.CLICommand.main(CLICommand.java:274)
  at hudson.cli.CliManagerImpl.main(CliManagerImpl.java:95)
  at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:498)
  at hudson.remoting.RemoteInvocationHandler$RPCRequest.perform(RemoteInvocationHandler.java:896)
  at hudson.remoting.RemoteInvocationHandler$RPCRequest.call(RemoteInvocationHandler.java:870)
  at hudson.remoting.RemoteInvocationHandler$RPCRequest.call(RemoteInvocationHandler.java:829)
  at hudson.remoting.UserRequest.perform(UserRequest.java:181)
  at hudson.remoting.UserRequest.perform(UserRequest.java:52)
  at hudson.remoting.Request$2.run(Request.java:336)
  at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:68)
  at hudson.cli.CliManagerImpl$1.call(CliManagerImpl.java:66)
  at hudson.remoting.CallableDecoratorAdapter.call(CallableDecoratorAdapter.java:18)
  at hudson.remoting.CallableDecoratorList$1.call(CallableDecoratorList.java:21)
  at jenkins.util.ContextResettingExecutorService$2.call(ContextResettingExecutorService.java:46)
  at java.util.concurrent.FutureTask.run(FutureTask.java:266)
  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
  at java.lang.Thread.run(Thread.java:748)
  Jul 20, 2017 9:56:38 AM hudson.remoting.JarCacheSupport$1 run
  WARNING: Failed to resolve a jar 0693b6b0ece47329e88e0f147bef2459
  java.io.IOException: Failed to write to /root/.jenkins/cache/jars/06/93B6B0ECE47329E88E0F147BEF2459.jar
  at hudson.remoting.FileSystemJarCache.retrieve(FileSystemJarCache.java:142)
  at hudson.remoting.JarCacheSupport$1.run(JarCacheSupport.java:64)
  at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
  at java.util.concurrent.FutureTask.run(FutureTask.java:266)
  at hudson.remoting.AtmostOneThreadExecutor$Worker.run(AtmostOneThreadExecutor.java:110)
  at java.lang.Thread.run(Thread.java:748)
  Caused by: java.io.IOException: Backing channel 'CLI connection to http://localhost:8080/' is disconnected.
  at hudson.remoting.RemoteInvocationHandler.channelOrFail(RemoteInvocationHandler.java:192)
  at hudson.remoting.RemoteInvocationHandler.invoke(RemoteInvocationHandler.java:257)
  at com.sun.proxy.$Proxy0.writeJarTo(Unknown Source)
  at hudson.remoting.FileSystemJarCache.retrieve(FileSystemJarCache.java:107)
  ... 5 more
  Caused by: hudson.remoting.Channel$OrderlyShutdown
  at hudson.remoting.Channel$CloseCommand.execute(Channel.java:1138)
  at hudson.remoting.Channel$1.handle(Channel.java:535)
  at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:83)
  Caused by: Command close created at
  at hudson.remoting.Command.<init>(Command.java:60)
  at hudson.remoting.Channel$CloseCommand.<init>(Channel.java:1132)
  at hudson.remoting.Channel$CloseCommand.<init>(Channel.java:1130)
  at hudson.remoting.Channel.close(Channel.java:1290)
  at hudson.remoting.Channel.close(Channel.java:1272)
  at hudson.remoting.Channel$CloseCommand.execute(Channel.java:1137)
```

--- a/libraries/plugin.rb
+++ b/libraries/plugin.rb
@@ -307,6 +307,8 @@ EOH
       path   = ::File.join(Chef::Config[:file_cache_path], "#{plugin_name}-#{version}.plugin")
       plugin = Chef::Resource::RemoteFile.new(path, run_context)
       plugin.source(source_url)
+      plugin.owner('jenkins')
+      plugin.group('jenkins')
       plugin.backup(false)
       plugin.run_action(:create)
 

